spot-contracts/contracts/RolloverVault.sol

```diff
@@ -739,80 +763,95 @@ contract RolloverVault is
 
         // Tranche as needed
-        (
-            uint256 underylingAmtToTranche,
-            uint256 seniorAmtToDeposit,
-            IBondController depositBond,
-            ITranche trancheIntoPerp
-        ) = perp.estimateUnderlyingAmtToTranche(s.perpTVL, perpAmtToMint);
-        _tranche(depositBond, underylingAmtToTranche);
+        IBondController depositBond = perp.getDepositBond();
+        ITranche trancheIntoPerp = perp.getDepositTranche();
+        (uint256 underylingAmtToTranche, uint256 seniorAmtToDeposit) = PerpHelpers.estimateUnderlyingAmtToTranche(
+            perpTVL,
+            perp.totalSupply(),
+            underlying_.balanceOf(address(depositBond)),
+            depositBond.totalDebt(),
+            trancheIntoPerp.totalSupply(),
+            seniorTR,
+            perpAmtToMint
+        );
+        _tranche(depositBond, underlying_, underylingAmtToTranche);
 
         // Mint perps
-        _checkAndApproveMax(trancheIntoPerp, address(perp), seniorAmtToDeposit);
+        _checkAndApproveMax(trancheIntoPerp, address(perp_), seniorAmtToDeposit);
         // NOTE: When the vault mints perps, it pays no fees.
-        perp.deposit(trancheIntoPerp, seniorAmtToDeposit);
+        perp_.deposit(trancheIntoPerp, seniorAmtToDeposit);
 
         // sync holdings
         _syncAndRemoveDeployedAsset(trancheIntoPerp);
-        _syncAsset(perp);
+        _syncAsset(perp_);
     }
 
     /// @dev Given a bond and its tranche data, deposits the provided amount into the bond
     ///      and receives tranche tokens in return.
     ///      Additionally, performs some book-keeping to keep track of the vault's assets.
-    function _tranche(IBondController bond, uint256 amount) private {
+    function _tranche(
+        IBondController bond,
+        IERC20Upgradeable underlying_,
+        uint256 underlyingAmt
+    ) private {
         // Skip if amount is zero
-        if (amount <= 0) {
+        if (underlyingAmt <= 0) {
             return;
         }
 
         // Get bond tranches
         BondTranches memory bt = bond.getTranches();
 
         // amount is tranched
-        _checkAndApproveMax(underlying, address(bond), amount);
-        bond.deposit(amount);
+        _checkAndApproveMax(underlying_, address(bond), underlyingAmt);
+        bond.deposit(underlyingAmt);
 
         // sync holdings
         for (uint8 i = 0; i < bt.tranches.length; i++) {
             _syncAndAddDeployedAsset(bt.tranches[i]);
         }
-        _syncAsset(underlying);
+        _syncAsset(underlying_);
     }
 
     /// @dev Rolls over freshly tranched tokens from the given bond for older tranches (close to maturity) from perp.
     ///      And performs some book-keeping to keep track of the vault's assets.
     /// @return Flag indicating if any tokens were rolled over.
-    function _rollover() private returns (bool) {
+    function _rollover(IPerpetualTranche perp_, IERC20Upgradeable underlying_) private returns (bool, ITranche) {
```

Comment:
Update @param and @return NatSpecs?

Reply:
It's a private function so I don't think we need @param or @notice. The `@return` here is just for clarity